src/pgen/cluster/agn_feedback.cpp

```diff
@@ -349,7 +362,32 @@ void AGNFeedback::FeedbackSrcTerm(parthenon::MeshData<parthenon::Real> *md,
                 "Kinetic injection leads to temperature below jet and existing gas");
 #endif
           }
+
+          // Apply velocity ceiling
+          const Real v2 =
+              SQR(prim(IV1, k, j, i)) + SQR(prim(IV2, k, j, i)) + SQR(prim(IV3, k, j, i));
+          if (v2 > vceil2) {
+            // Fix the velocity to the velocity ceiling
+            const Real v = sqrt(v2);
+            cons(IM1, k, j, i) *= vceil / v;
+            cons(IM2, k, j, i) *= vceil / v;
+            cons(IM3, k, j, i) *= vceil / v;
+            prim(IV1, k, j, i) *= vceil / v;
+            prim(IV2, k, j, i) *= vceil / v;
+            prim(IV3, k, j, i) *= vceil / v;
+
+            // Update the internal energy
```

Comment:
For the comment: Update total? Or "remove kinetic energy"?

Reply:
Ya, we're not changing the internal energy here, we're removing the kinetic energy from the total energy